sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/TemplatingDataflowPipelineRunnerTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.dataflow.sdk.runners;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.options.DataflowPipelineOptions;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.options.PipelineOptionsFactory;
+import com.google.cloud.dataflow.sdk.testing.ExpectedLogs;
+import com.google.cloud.dataflow.sdk.testing.TestDataflowPipelineOptions;
+import com.google.cloud.dataflow.sdk.util.NoopPathValidator;
+import com.google.cloud.dataflow.sdk.util.TestCredential;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.File;
+
+/**
+ * Tests for TemplatingDataflowPipelineRunner.
+ */
+@RunWith(JUnit4.class)
+public class TemplatingDataflowPipelineRunnerTest {
+
+  @Mock private DataflowPipelineJob mockJob;
+
+  @Rule
+  public ExpectedLogs expectedLogs = ExpectedLogs.none(TemplatingDataflowPipelineRunner.class);
+
+  @Rule
+  public ExpectedException expectedThrown = ExpectedException.none();
+
+  @Rule
+  public final TemporaryFolder tmpDir = new TemporaryFolder();
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    when(mockJob.getProjectId()).thenReturn("project");
+    when(mockJob.getJobId()).thenReturn("job");
+  }
+
+  /**
+   * Returns a {@link TemplatingDataflowPipelineRunner} that will return the provided a job to
+   * return. Some {@link PipelineOptions} will be extracted from the job, such as the project ID.
+   */
+  private TemplatingDataflowPipelineRunner createMockRunner(
```

Comment:
I think you could use the MockitoAnnotations to create the mock runner as well, and do more initialization in setup.